testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java

```diff
@@ -41,8 +50,30 @@
     }
   }
 
+  @BeforeAll
+  public static void beforeAll(StargateEnvironmentInfo stargate) {
+    STARGATE_ADDRESSES =
+        stargate.nodes().stream()
+            .map(
+                n -> {
+                  try {
+                    return InetAddress.getByName(n.seedAddress());
+                  } catch (UnknownHostException e) {
+                    throw new RuntimeException(e);
+                  }
+                })
+            .collect(Collectors.toList());
+  }
+
   @BeforeEach
   public void init(ClusterConnectionInfo backend) {
     this.backend = backend;
   }
+
+  // TODO generalize this to an ExecutionCondition that reads custom annotations, like
+  // @CassandraRequirement/@DseRequirement in the Java driver tests
+  public boolean isCassandra4() {
+    return !backend.isDse()
+        && Version.parse(backend.clusterVersion()).nextStable().compareTo(Version.V4_0_0) >= 0;
+  }
```

Comment:
Same as above, this used to be on `JavaDriverTestBase`. There's nothing driver-specific here actually.
It should be possible to bake support for `@CassandraRequirement` / `@DseRequirement` directly into `StargateContainer`, we don't even need to redefine them really.